### PR TITLE
[MM-60422] Add GetChannelsWithActivityDuring

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- [modify GetUsersInChannelDuring to accept a slice of channelIds](https://github.com/mattermost/mattermost/commit/e644d9c2064f778bd8bc8bc0be80533ce885f207)
- [add GetChannelsWithActivityDuring](https://github.com/mattermost/mattermost/commit/106458a1866d72e75d578fa975e5d7f6e95391e3)

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-60422

#### Release Note
```release-note
GetUsersInChannelDuring now accepts a slice; added GetChannelsWithActivityDuring
```

### CODE REVIEW

File: server/channels/store/sqlstore/channel_member_history_store.go [hunk 1]
```diff
@@ -65,27 +65,78 @@ func (s SqlChannelMemberHistoryStore) LogLeaveEvent(userId string, channelId str
 	return nil
 }
 
-func (s SqlChannelMemberHistoryStore) GetUsersInChannelDuring(startTime int64, endTime int64, channelId string) ([]*model.ChannelMemberHistoryResult, error) {
+func (s SqlChannelMemberHistoryStore) GetChannelsWithActivityDuring(startTime int64, endTime int64) ([]string, error) {
```
Comment (agnivade) [inline server/channels/store/sqlstore/channel_member_history_store.go:68]: I see that you added an app layer test, but typically for every store method, we also add a store layer test.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by agnivade — APPROVED** [review]

**Comment by hmhealey** [comment]
@cpoile Could you add review labels to this PR so that we know where it's at?

**Comment by cwarnermm** [comment]
@cpoile - Are there product, developer, and/or API changes needed for this Engineering PR?

**Comment by cpoile** [comment]
@cwarnermm It's an internal api change, there's nothing external-facing. I think we're ok with no new docs.

**Pull request merged**